synapse/federation/sender/__init__.py

```diff
@@ -355,6 +355,39 @@ async def handle_event(event: EventBase) -> None:
                     if not is_mine and send_on_behalf_of is None:
                         return
 
+                    # We also want to not send out-of-band membership events.
+                    #
+                    # OOB memberships are used in three (and a half) situations:
+                    #
+                    # (1) invite events which we have received over federation. Those
+                    #     will have a `sender` on a different server, so will be
+                    #     skipped by the "is_mine" test above anyway.
+                    #
+                    # (2) rejections of invites to federated rooms. These are normally
+                    #     created via federation (in which case the remote server is
+                    #     responsible for sending out the rejection). If that fails,
+                    #     we'll create a leave event locally, but that's only really
```

Comment:
And it's this local fallback event that is considered out-of-band?

Reply:
I like the rework. But the underlying difficulty I'm having here is that I don't grok the meaning of `out of band`. It makes me think of an event that has been received by this server over some non-Matrix protocol, e.g. carrier pigeon.

Reply:
it is maybe not the best term (see https://github.com/matrix-org/synapse/pull/4405#pullrequestreview-196184944, which is where we expanded the idea beyond case (1).)

It's "out of band" inasmuch as it's not sent over the regular `federation/v1/send` API.

Reply:
There are only two hard problems, after all.

Reply:
I've updated the wording here in ca5b502bb8.